internals/overlord/servstate/manager_test.go

```diff
@@ -297,6 +297,53 @@ services:
 	c.Assert(s.manager.StopTimeout(), Equals, time.Minute*60+time.Millisecond*200)
 }
 
+func (s *S) TestStopServiceWithinOkayDelay(c *C) {
+	// A longer okayDelay is used so that the change for starting the service won't
+	// quickly transition into the running state.
+	fakeOkayDelay := 5 * shortOkayDelay
+	servstate.FakeOkayWait(fakeOkayDelay)
+
+	s.newServiceManager(c)
+	// The service sleeps for fakeOkayDelay second then creates a side effect (a file at donePath).
+	layer := `
+services:
+    %s:
+        override: replace
+        command: /bin/sh -c "sleep %g; {{.NotifyDoneCheck}}"
+`
+	serviceName := "test-stop-within-okaywait"
+	s.planAddLayer(c, fmt.Sprintf(layer, serviceName, fakeOkayDelay))
```

Comment:
I think this is buggy, as `fakeOkayDelay` is an integer `time.Duration` and can't be formatted with `%g`. I think we want `fakeOkayDelay.Seconds()`. I think the test was erroneously passing as a result.

Nit: I think it's easier to read to have the `fmt.Sprintf(...)` call above on the `layer :=` line, so the sprintf values are more closely associated with the format string.

Reply:
`fakeOkayDelay.Seconds()` (together with the next comment) is fixed.

On the readability issue, I do agree if we have something like the following, it will be more readable:

```go
layer := `some template`
layer := fmt.Sprintf(layer, foo, bar)
s.planAddLayer(c, layer)
```

However, in the current test file, there are many test cases with the following code:

```go
layer := `some template`
s.planAddLayer(c, fmt.Sprintf(
    layer,
    foo, 
    bar
))
```

So, for consistency, I used the same code (I will move `serviceName` above `layer`, so that this part is exactly the same code organization as other test cases).